src/sentry/api/event_search.py

```diff
@@ -304,13 +307,10 @@ def _flatten(seq):
             else:
                 yield item
 
-    if not (children and isinstance(children, list) and isinstance(children[0], list)):
-        return children
```

Comment:
this was buggy: a list which only had nested values further on would not get flattened.  this only happened inside the paren expression as far as I can tell